frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card.tsx

```diff
@@ -18,43 +21,73 @@ const TopConsumerResourceValue = {
   [PROJECTS]: 'PROJECTS_',
   [STORAGE_CLASSES]: 'STORAGE_CLASSES_',
   [PODS]: 'PODS_',
-  [VMS]: 'VMS_',
 };
 const TopConsumerSortByValue = {
   [BY_USED]: 'BY_USED',
   [BY_REQUESTED]: 'BY_REQUESTED',
 };
-
 const TopConsumerResourceValueMapping = {
   Projects: 'namespace',
   'Storage Classes': 'storageclass',
   Pods: 'pod',
 };
 
-const metricTypes = _.keys(TopConsumerResourceValue);
-const sortByTypes = _.keys(TopConsumerSortByValue);
+const getTopConsumersQueriesMap = (
+  pluginItem,
+  topConsumers,
+  category,
+): getTopConsumersQueriesMapType => {
+  const { name } = pluginItem.properties;
+  const { queryName } = pluginItem.properties;
+  if (!topConsumers[queryName]) {
+    TopConsumerResourceValue[name] = _.replace(queryName, category, '');
+    TopConsumerResourceValueMapping[name] = pluginItem.properties.metricType;
+    topConsumers[queryName] = pluginItem.properties.query;
+  }
+  return topConsumers;
+};
+const getItems = (plugin: any, flags: FlagsObject) =>
+  plugin.filter((e) => flags[e.properties.required]);
 
-const metricTypesOptions = _.zipObject(metricTypes, metricTypes);
-const sortByOptions = _.zipObject(sortByTypes, sortByTypes);
+const getTopConsumersQueries = (flags: FlagsObject) => {
+  let topConsumers: any = { ...TOP_CONSUMER_QUERIES };
+  getItems(plugins.registry.getDashboardsStorageTopConsumerUsed(), flags).forEach((pluginItem) => {
+    topConsumers = getTopConsumersQueriesMap(pluginItem, topConsumers, 'BY_USED');
+  });
+  getItems(plugins.registry.getDashboardsStorageTopConsumerRequested(), flags).forEach(
+    (pluginItem) => {
+      topConsumers = getTopConsumersQueriesMap(pluginItem, topConsumers, 'BY_REQUESTED');
+    },
+  );
+  return topConsumers;
+};
 
-const TopConsumerCard: React.FC<DashboardItemProps> = ({
+const TopConsumerCard: React.FC<DashboardItemProps & WithFlagsProps> = ({
   watchPrometheus,
   stopWatchPrometheusQuery,
   prometheusResults,
+  flags = {},
 }) => {
+  const metricTypes = _.keys(TopConsumerResourceValue);
+  const sortByTypes = _.keys(TopConsumerSortByValue);
+  const metricTypesOptions = _.zipObject(metricTypes, metricTypes);
+  const sortByOptions = _.zipObject(sortByTypes, sortByTypes);
+
   const [metricType, setMetricType] = React.useState(metricTypes[0]);
   const [sortBy, setSortBy] = React.useState(sortByTypes[0]);
   React.useEffect(() => {
+    const topConsumers = getTopConsumersQueries(flags);
     const query =
-      TOP_CONSUMER_QUERIES[
+      topConsumers[
         StorageDashboardQuery[TopConsumerResourceValue[metricType] + TopConsumerSortByValue[sortBy]]
```

Comment:
`StorageDashboardQuery` wont have VMS_BY_PODS etc. Should this actually be just following ?
```suggestion
        TopConsumerResourceValue[metricType] + TopConsumerSortByValue[sortBy]
```

Same for the one below